PointerStar/Client/ViewModels/RoomViewModel.cs

```diff
@@ -69,14 +74,36 @@ public bool IsTeamMember
         }
     }
 
-    public RoomViewModel(IRoomHubConnection roomHubConnection, ICookie cookie, HttpClient httpClient)
+    public RoomViewModel(IRoomHubConnection roomHubConnection, ICookie cookie, HttpClient httpClient, IClipboardService clipboardService)
     {
         RoomHubConnection = roomHubConnection ?? throw new ArgumentNullException(nameof(roomHubConnection));
         Cookie = cookie ?? throw new ArgumentNullException(nameof(cookie));
         HttpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+        ClipboardService = clipboardService ?? throw new ArgumentNullException(nameof(clipboardService));
+
         RoomHubConnection.RoomStateUpdated += RoomStateUpdated;
     }
 
+    public async Task OnClickClipboard(object sender, string? url)
+    {
+        if (!string.IsNullOrEmpty(url))
+        {
+            ClipboardService?.CopyToClipboard(url);
```

Comment:
Services are null checked in the constructor so you don't have to null check them everywhere else.
```suggestion
            await ClipboardService.CopyToClipboard(url);
```